Card/Token/index.ts

```diff
@@ -0,0 +1,158 @@
+import { cryptly } from "cryptly"
+import { isoly } from "isoly"
+import { isly } from "isly"
+import { Expires } from "../Expires"
+import { Month } from "../Expires/Month"
+import { Year } from "../Expires/Year"
+import type { Card } from "../index"
+import { Masked } from "../Masked"
+import { Part } from "../Part"
+import { Key as TokenKey } from "./Key"
+import { Type } from "./Type"
+import { Unpacked as TokenUnpacked } from "./Unpacked"
+
+type CoreToken = `${string}/${string}/${string}/${string}/${string}/${string}`
+export type Token = CoreToken | `${CoreToken}/${Part}`
+export namespace Token {
+	export import Key = TokenKey
+	const patternString =
+		"(\\d{10,12})\\/(\\d{2})\\/([0-1]\\d{3})\\/((RSA[A-Za-z]{4})|([0-2]\\d[0-6][02468])|(((([02468][048])|([13579][2,6]))(02((0[1-9])|([1-2]\\d))))|(\\d{2}((((0[13578])|(1[02]))((0[1-9])|([1-2]\\d)|(3[01])))|(02((0[1-9])|(1\\d)|(2[0-8])))|(((0[469])|(11))((0[1-9])|([1-2]\\d)|(30)))))))\\/[a-zA-Z0-9\\-_]+\\/[a-zA-Z0-9\\-_]+(\\/(pan|csc|expires|month|year|masked))?"
+	export const pattern = new RegExp("^" + patternString + "$")
+	export const globalPattern = new RegExp(patternString, "g")
+	export const type = isly.named("pax2pay.cde.Card.Token", isly.string<Token>(pattern))
+	export const is = type.is
+	export const flaw = type.flaw
+	//long term token regexp 6 length
+	//^(((([02468][048])|([13579][2,6]))(02((0[1-9])|([1-2][0-9]))))|(\d{2}((((0[13578])|(1[02]))((0[1-9])|(1|2[0-9])|(3[01])))|(02((0[1-9])|(1[0-9])|(2[0-8])))|(((0[469])|(11))((0[1-9])|([1-2][0-9])|(30))))))$
+
+	// masked/length/expires/key/encrypted/salt
+	// masked/length/expires/key/encrypted/salt/part
+	// 4567897890/16/0221/1354/0ktG52FXmULx7-3mrj0smEWvJWwuJNA9eQNr8O8kBBKy_gvg/FlBUNQjpk4R9g_dcw6WYzQ
+	// 4567897890/16/0221/1354/0ktG52FXmULx7-3mrj0smEWvJWwuJNA9eQNr8O8kBBKy_gvg/FlBUNQjpk4R9g_dcw6WYzQ/csc
+	// 4567897890/16/0221/220101/0ktG52FXmULx7-3mrj0smEWvJWwuJNA9eQNr8O8kBBKy_gvg/FlBUNQjpk4R9g_dcw6WYzQ
+	export function getType(value: string): Type | undefined {
+		return !is(value) ? undefined : unpack(value).key.startsWith("RSA") ? "asymmetric" : "symmetric"
+	}
+	export import Unpacked = TokenUnpacked
+	export function unpack(token: Token): Unpacked
+	export function unpack(
+		masked: string,
+		length: string,
+		expires: string,
+		key: string,
+		value: string,
+		salt: string
+	): Unpacked
+	export function unpack(
+		masked: string,
+		length: string,
+		expires: string,
+		key: string,
+		value: string,
+		salt: string,
+		part: string
+	): Unpacked
+	export function unpack(token: string[]): Unpacked
+	export function unpack(...argument: Token[] | string[] | string[][]): Unpacked {
```

Comment:
export function unpack(...argument: [Token | string[]] | string[]): Unpacked {

Reply:
you can define array types with [string]? weird, like how in c etc you can write []char but if you do you wont see the light of heaven

Reply:
`string[]` is an array, while `[string]` is a [tuple](https://www.typescriptlang.org/play/?#code/PTAEBUE8AcEsGMCGAbZlSIHYYE48evAPaYAuismAzqAF4CmORopzAtlukQEYBW98UjSIAzDACgQoKpQDmyeixj0AdBGUBleDljRSoABaIaVaANgoMmFJBk1ETAK6YAJpLAP8t0AHcDCA1BiMgpsNkdkUl0FJTMqABorF18DRkVSVNAiHBdGd1BKFICWVPQHRUpcgA96ZNgaWDZobPIyFXF88FSqRXKglAVk0kdoBSo1LsoAayyxDPo2DHtfAhYifODMAX0qIjZFF0RyRO5HfR9YDNAxk0gQqpKsUCn6SFqsvm3xjqkATSJHP1sNp6EdeiwRjFHDJMLJQAApRAAN0QWh0egA5PY8KsqHdyFUAFwdYJUfQiCjIOQAJXophIPVAAF5QABtABEADkiPoAGIA1zsxIAFgADMKALoAbh+YFO+kgAN8sFQoC271YoFy8GQfUuNFIyiWGAho1UJIZ+mgxhhslp9Oo9EJbLJOlhiUwjjY3EYEuZbPZAG8AL5C0AAJlFoulstAAEkxIrAQYiEjGFk0zgSukfMwUTpENwYtZ9jQk0DpPR0ql8i5YCIRGlMPBFD7SD4q9gnp5VgAKUC911yUAAHzVXp9OAAlKy-VOkvl5qaYv3WUP3ePvb7QFP2vkAIL2Ls4yCJJfZXJZ+pqnkFJotLA7ZhPS4LDCkfKcAquegPYgRZIfVAehLlSLNEGkUg3ThbITU9LccDUONsHmRdIWrRQL3TLBkgUWErj6WRHAcR8qxcPd6wHa0qFte1mkdVkAEY-SZViIyjedA3EUAgkta4iCQZBkJEZgWXhDQAHlORUa0cB6XtqNoul6J6VloynGUeNJIgFBUZAiFkXt9ME4SiA08Rg1jSYaH2LAaCgMw0V0c4VWQUBoCYJFYFybNeLwbZYjpd9F0yHRZAMfRKl-RIcOAtNsHwepemwRgmCvRMlSg9BWE-eAWxoqwPn4QR30K5wAFptV1HB3iiqo90UmllIZehWXDGMD2XRQkGwRt6DcqkXhNWQiCIZJrVIUhGF62CqBTHB9G4fVZg2EgtkEd5DnILIsxE3bYCqYYau+cRDTMUANHIBt9zygV9BZVl4MnRJ11kF6oLkd7oIAfg60kdiukRnUuxBrtu5xSFnf1WW4tlRUSdl90OTAplzMN2UQZHUaIFR6AAAXZCV4lhpiEYAaSwIhUawdGpkp6nMBUKgCaJknwwR-ctkcN10awehucoJmCYRjQRkYLzdhwQniY6qQAHVUmwcsjDTE0en0URnkwIgfBQ5R7H0JcfVkShMGHTXEDQs0kmzLtsGcKZtd1656HwgxEnLHrQGhDDpA80E3CkIgzHwVgs01SoEDBR5Iv0L2VeS9A8NkAjXGzfJfygiC6qCi5VSAi33IcKJ4AiBxAuJU6jQABQIS7HG4GgHpBsH-zIRIVE7p7GFnP6+OtSB68b51a8H4ZG6hh6SbJUGRDUomIwAVmjYmeLXQGmIX8Nl8ScMADYV+njeV9AABmKNEnP+Gz6jVm++ofQ2BIDIJK2Uf9jIJui7HhuqHnzfQAAGpv5Dz-sxABwCB6gLahKTeMp-qgCfmQAw4Bczv1dkIf0UDx5gNge1IBICcEwJgfAviSCMhdBqvQdBn8sF1yIe1EhsZ-iAi9j7LqBpmC5CoNoWAQERDOEELABkRR4CBHIC8T89sfwiEoO8buWZNayUQPsKacllRXDOnSSuVU+gCObFEEg-RkCl11FNUe-IcAAFEmj6TePQXs3lnQKI7p3BwsgqDOlZJ3FQCje5TmcRORgMpxCCVMWCCx2QbGjCIPYwcx9YEn2wb-f+zFzJhLLuYggljol2KrPE2em956JGSRPcBzESn0JSeA9q5l3D5AiqQaAniQA+DaSoLRPD0SkF1LCFQ2RZDABcAJKgwAjCuG4KNKYwAaoKGMPQcq2spqjM6bwvQ5VT7lVFCoCKbBkAAGJhhmioOVSg5VjqkHKso1RjATk4XKqYGqmNyq-n9jRYR1AGmTWaYSEAiAzgGG2cQNgwAiwGWAKs7pGyXlVBidBcqRyxgItSOVB2TtMAIuUMAcQQA) containing a string at index 0, so not quite the same